File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/mesh/persistence/max3d/Max3dsModelData.java
package com.abubusoft.xenon.mesh.persistence.max3d;

/**
 * <p>Sempre e solo triangoli</p>
 * @author Francesco Benincasa
 *
 */
public class Max3dsModelData {

	public String name;
	/**
	 * startX 3
	 */
	public float[] vertices;
	/**
	 * startX 3
	 */
	public float[] normals;
	/**
	 * startX 2
	 */
	public float[] textCoords;
	/**
	 * startX 3
	 */
	public int[] indices;

	public Max3dsModelData(String value) {
		name=value;
	}

	public void setData(float[] aVertices, float[] aNormals, float[] aTexCoords, int[] aIndices) {		
		vertices=aVertices;
		normals=aNormals;
		textCoords=aTexCoords;
		indices=aIndices;
		
	}

}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/mesh/persistence/max3d/Max3dsModelData.kt
package com.abubusoft.xenon.mesh.persistence.max3d

/**
 *
 * Sempre e solo triangoli
 * @author Francesco Benincasa
 */
class Max3dsModelData(var name: String) {
    /**
     * startX 3
     */
    var vertices: FloatArray

    /**
     * startX 3
     */
    var normals: FloatArray

    /**
     * startX 2
     */
    var textCoords: FloatArray

    /**
     * startX 3
     */
    var indices: IntArray
    fun setData(aVertices: FloatArray, aNormals: FloatArray, aTexCoords: FloatArray, aIndices: IntArray) {
        vertices = aVertices
        normals = aNormals
        textCoords = aTexCoords
        indices = aIndices
    }
}